### ISSUE TITLE
[rush] Git Config Not Fully Evaluated (allowedEmailRegExps)

### ISSUE THREAD

**Issue by btilford** [microsoft/rushstack#1861]
## Please prefix the issue title with the project name i.e. [rush], [api-extractor] etc. ##

**Is this a feature or a bug?**

- [ ] Feature
- [X] Bug

**Please describe the actual behavior.**
So I'll admit I've got a fairly complex .gitconfig setup but it is valid. I'll describe that first.

For my **~/.gitconfig** I have it in a repo that I can quickly add to any machine I need to use git on. I've kept user info out of this but from it I include another git config that is per machine. **~/.gitconfig.local**
```
[user]
        email = blank@blank.com
        name = btilford
        signingKey = ***

[include]
    path = ~/.gitconfig.local
```

Within **~/.gitconfig.local** I have a default (personal) user.\[name,email] set and then conditionaly include project/company specific git configs based on path

```
[includeIf "gitdir:~/work/company-a/"]
    path = ~/work/company-a/.gitconfig
[includeIf "gitdir:~/work/company-b/"]
    path = ~/work/company-b/.gitconfig

```
And then inside those company specific files I setup credentials for each company and override anything else I need to.

When evaluating the user email regex rush will only see the default from **~/.gitconfig.local** and not the ones included conditionally which causes a build error.

**What is the expected behavior?**
Rush should evaluate git configuration exactly like git itself does.

**If this is a bug, please provide the tool version, Node.js version, and OS.**

* **Tool:** Rush 
* **Tool Version:**  5.23.0
* **Node Version:** 12,13,14
  * **Is this a LTS version?** 
  * **Have you tested on a LTS version?** yes
* **OS:** Linux 5.3.0-51-generic #44-Ubuntu SMP Wed Apr 22 21:09:44 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

**Comment by iclanton** [comment]
Interesting. IIRC, we just use `git` commands to determine what the configured email is. Maybe we're passing a flag to it that is only scoping to a the `.local` file.

**Comment by btilford** [comment]
Yeah, so it looks like restricting git config scope would do that, technically all my configs are in the global scope but if you pass either `--local` or `--global` when looking up a value you get nothing. Interesting thing is `git config --show-scope user.email` finds the value and shows the global scope.

```console
cd company1
git config --get user.email
> ben@company1.com
git config --show-scope user.email
> global  ben@company1.com
git config --global --get user.email
>
git config --local --get user.email
>
```
